components/concept/engine/src/main/java/mozilla/components/concept/engine/webnotifications/WebNotification.kt

```diff
@@ -7,31 +7,23 @@ package mozilla.components.concept.engine.webnotifications
 /**
  * A notification sent by the Web Notifications API.
  *
- * @property origin The website that fired this notification.
  * @property title Title of the notification to be displayed in the first row.
- * @property body Body of the notification to be displayed in the second row.
  * @property tag Tag used to identify the notification.
- * @property iconUrl Medium image to display in the notification.
+ * @property text Body of the notification to be displayed in the second row.
+ * @property imageUrl Medium image to display in the notification.
  * Corresponds to [android.app.Notification.Builder.setLargeIcon].
- * @property vibrate Vibration pattern felt when the notification is displayed.
+ * @property textDirection Preference for text direction.
+ * @property lang language of the notification.
+ * @property requireInteraction Preference flag that indicates the notification should remain.
  * @property timestamp Time when the notification was created.
- * @property requireInteraction Preference flag that indicates the notification should remain
- * active until the user clicks or dismisses it.
- * @property silent Preference flag that indicates no sounds or vibrations should be made.
- * @property onClick Callback called with the selected action, or null if the main body of the
- * notification was clicked.
- * @property onClose Callback called when the notification is dismissed.
  */
 data class WebNotification(
-    val origin: String,
-    val title: String? = null,
-    val body: String? = null,
-    val tag: String? = null,
-    val iconUrl: String? = null,
-    val vibrate: LongArray = longArrayOf(),
-    val timestamp: Long? = null,
-    val requireInteraction: Boolean = false,
-    val silent: Boolean = false,
-    val onClick: () -> Unit,
-    val onClose: () -> Unit
+    val title: String?,
+    val tag: String,
+    val text: String?,
+    val imageUrl: String?,
```

Comment:
The [Notification spec](https://developer.mozilla.org/en-US/docs/Web/API/Notification/Notification) has both "image" and "icon". Icon refers to the medium image, but "image" refers to big picture style notifications.

Reply:
Yes, I agree.  But currently GV only gives us a imageURL for the icon.  No other information.
Here's the GV's `WebNotification` spec https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/WebNotification.java

Reply:
I wasn't clear in my comment but I'd prefer to use `iconUrl` as the property name here. I previous discussed the GV version with Alvina and the intent is that it represents the "icon" field.

Reply:
I see.  Currently we don't have a icon URL from GV though.  We only get the image URL so that's the image when you expand the notification.  So my implementation can only allow the browser to specify the icon.  In the future when GV pass us the icon URL we can then update this class and use it.

Reply:
GV's `imageUrl` is actually the `icon` field of a web notification. It represents the medium 48x48 icon rather than the expanded big picture image. GV doesn't give us the `image` field right now.

Reply:
I see. I was thinking the image that we put in the large icon is the image.  But looking into this further I see that there's another image (a much bigger image) that we can show in the notification.  Sure, I'll rename it to `iconUrl`  Thanks,